lib/workers/repository/config-migration/branch/migrated-data.ts

```diff
@@ -1,15 +1,56 @@
 import detectIndent from 'detect-indent';
 import JSON5 from 'json5';
+import prettier from 'prettier';
 import { migrateConfig } from '../../../../config/migration';
 import { logger } from '../../../../logger';
 import { readLocalFile } from '../../../../util/fs';
+import { getFileList } from '../../../../util/git';
 import { detectRepoFileConfig } from '../../init/merge';
-
 export interface MigratedData {
   content: string;
   filename: string;
 }
 
+async function applyPrettierFormatting(
+  content: string,
+  fileName: string
+): Promise<string> {
+  const prettierConfigFilenames = [
+    '.prettierrc',
+    '.prettierrc.json',
+    '.prettierrc.yml',
+    '.prettierrc.yaml',
+    '.prettierrc.json5',
+    '.prettierrc.js',
+    '.prettierrc.cjs',
+    'prettier.config.js',
+    'prettier.config.cjs',
+    '.prettierrc.toml',
+  ];
+  let prettierExists = (await getFileList()).some((file) =>
+    prettierConfigFilenames.includes(file)
+  );
+  const packageJsonContent = await readLocalFile('package.json', 'utf8');
+  try {
+    prettierExists ||=
+      packageJsonContent && JSON.parse(packageJsonContent).prettier;
+  } catch {
+    logger.warn('Invalid JSON found in package.json');
+  }
+
+  if (!prettierExists) {
+    return content;
+  }
+
+  let newContent = content;
+
+  newContent = prettier.format(content, {
+    filepath: fileName,
+  });
+
+  return newContent;
```

Comment:
```suggestion
  return prettier.format(content, {
    filepath: fileName,
  });
```

Can prettier ever throw, or return null? We should use the original content if any error occurs

Reply:
It only returns string, and incase of errors it returns the original string.